### ISSUE TITLE
variational_autoencoder_deconv.py |  CustomVariationalLayer warning while compiling and error while training

### ISSUE THREAD

**Issue by sedflix** [keras-team/keras#8824]
This is with reference to [examples/variational_autoencoder_deconv.py](https://github.com/keras-team/keras/blob/master/examples/variational_autoencoder_deconv.py)

While compiling the model, I get this warning:

    /usr/local/lib/python3.5/dist-packages/ipykernel_launcher.py:119: UserWarning: Output 
    "custom_variational_layer_1" missing from loss dictionary. We assume this was done on purpose, and 
    we will not be expecting any data to be passed to "custom_variational_layer_1" during training.

While trying to train [ model.fit() ], I get the following error:

```
InvalidArgumentError (see above for traceback): Incompatible shapes: [235200] vs. [1215000]
	 [[Node: custom_variational_layer_1/logistic_loss/mul = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](custom_variational_layer_1/Log, custom_variational_layer_1/Reshape)]]
	 [[Node: loss/add/_153 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_1446_loss/add", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

```
See the gist for detailed stack trace of the error. 
[Link to code(a gist)]( https://gist.github.com/geekSiddharth/9d23bedc8f6370b5578067757af11463)

Using: 
keras==2.1.2
tensorflow-gpu==1.4.1
Cuda compilation tools, release 7.5, V7.5.17
CUDNN_MAJOR      6
CUDNN_MINOR      0
 GK210GL [Tesla K80]
